### ISSUE TITLE
[vue-tsc] Cannot find module 'App.vue' or its corresponding type declarations.

### ISSUE THREAD

**Issue by Jnp95** [vuejs/language-tools#4082]
In all bun projects, including [this basic template](https://github.com/sistematico/bun-vite-template), the command `vue-tsc` produces this error:
 ```
src/main.ts:3:17 - error TS2307: Cannot find module './App.vue' or its corresponding type declarations.

3 import App from './App.vue'
                  ~~~~~~~~~~~


Found 1 error in src/main.ts:3
```

Whereas it's ok both for linter and static build.
Is there a specific missing setup for bun (opposed to Node.js)?

**Comment by manshen666** [comment]
You can roll back Vue - Official version to 1.8.27
![image](https://github.com/vuejs/language-tools/assets/80823786/90e8cdf7-41a5-4599-8971-a739fffa945f)

**Comment by Jnp95** [comment]
> You can roll back Vue - Official version to 1.8.27

Actually, VSCode and extensions should not be concerned : I should be able to use `vue-tsc` command independently (on a pipeline for example, without any EDI).

**Comment by Jnp95** [comment]
> Does other package manager / runtime, such as npm and node.js, have the same error?

No, it doesn't! Which is very surprising! `Bun` _might_ be identified as the responsible.

**Comment by nethriis** [comment]
I have the same error on node.js

**Comment by Jnp95** [comment]
@nethriis 
> I have the same error on node.js

[This example](https://stackblitz.com/edit/vitejs-vite-c2jqqb?file=src%2FApp.vue) works fine with Node.js / Vue.js / Vite.
Use: `npm run type-check` to detect my introduced errors in `App.vue` file.

**Comment by nethriis** [comment]
@Jnp95 
> npm run type-check

Doesn't work for me, I don't know why, until now it worked fine (bizarre)

**Comment by johnsoncodehk** [comment]
vue-tsc cannot work in Bun because it requires modifying the node's fs module, which Bun does not allow.

https://github.com/volarjs/volar.js/blob/0f861d59faa19cbeadef182ee079be88f9629fc7/packages/typescript/lib/quickstart/runTsc.ts#L18

**Comment by Jnp95** [comment]
[Suggestion from here](https://github.com/oven-sh/bun/issues/4754#issuecomment-2011949253):

> Most likely the linked issue means that overriding readFileSync as a way to hook into module loading. We do not support that currently.

> We do support two alternative approaches though:
> - `require` can be overriden and we have code in place to fully support that
> - We have pretty good support for [Loaders](https://bun.sh/docs/runtime/plugins#loaders) that work for both ESM and CJS (unlike overriding require)

**Comment by davidmatter** [comment]
Let's use this one to discuss bun support as a feature request instead of a bug :)

**Comment by theotheroracle** [comment]
i think bun should probably be removed from the quickstart guide if it's not supported, as i was very confused why the demo project was broken

**Comment by johnsoncodehk** [comment]
Please track https://github.com/oven-sh/bun/issues/4754.

**Comment by seepine** [comment]
I think bun should probably be removed from the [quickstart](https://vuejs.org/guide/quick-start.html) guide if it's not supported, as i was very confused why the demo project was broken.


There appears to be an open PR with no one working on it.
https://github.com/volarjs/volar.js/pull/310

---

Progress update:

Temporary workaround before this [volar#310 PR](https://github.com/volarjs/volar.js/pull/310) is merged

```sh
bun rm vue-tsc
bun add @seepine/vue-tsc -D

# it can work now
bunx -bun vue-tsc -b --noEmit
```

> @seepine/vue-tsc: https://github.com/seepine/vue-tsc
